app/webpacker/controllers/internal_events_controller.js

```diff
@@ -0,0 +1,47 @@
+import { Controller } from 'stimulus';
+
+export default class extends Controller {
+  static targets = ['name', 'startAt', 'partialUrl'];
+
+  get nameValue() {
+    return this.nameTarget.value;
+  }
+
+  get startAtValue() {
+    return this.startAtTarget.value;
+  }
+
+  get partialUrlValue() {
+    return this.partialUrlTarget.value;
+  }
+
+  set partialUrlValue(value) {
+    this.partialUrlTarget.value = value;
+  }
+
+  generateButtonClicked(event) {
+    event.preventDefault();
+
+    const date = this.formatDate(this.startAtValue);
+    const name = this.formatName(this.nameValue);
+    this.partialUrlValue = this.generatePartialUrl(date, name);
+  }
+
+  generatePartialUrl(date, name) {
+    if (date === '' || name === '') return '';
+
+    return `${date}-${name}`;
+  }
+
+  formatDate(dateTimeString) {
+    if (dateTimeString === '') return '';
+
+    const date = dateTimeString.substring(0, dateTimeString.indexOf('T'));
+    const [year, month, day] = date.split('-');
+    return year.slice(-2) + month + day;
```

Comment:
This is pretty hard to follow - would it be cleaner to construct a new `Date` from the string and then output that in a format?

Reply:
Pete made the same comment, not sure if there is a way to do this.

Reply:
Yeah it's a bit of a faff because JavaScript's date functionality is pretty limited. [toLocaleString](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleString) has lots of options but it's a total faff compared to `#strftime`.

```suggestion
    const date = new Date(dateTimeString);
    return date.toLocaleString('en-GB', { '🤷🏽‍♂️' });
```